### PR TITLE
Update messaging fo the after merge comments

### PR DESCRIPTION
**Pull request by buyaa-n**
Improve messaging, addressing feedback [in the PR](https://github.com/dotnet/roslyn-analyzers/pull/6153#discussion_r977126738)

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx [hunk 1]
```diff
@@ -1990,7 +1990,7 @@
     <value>Starting with .NET 7 the operator '{0}' will throw when overflowing in a checked context. Wrap the expression with an 'unchecked' statement to restore the .NET 6 behavior.</value>
   </data>
   <data name="PreventNumericIntPtrUIntPtrBehavioralChangesTitle" xml:space="preserve">
-    <value>Prevent from behavioral change</value>
+    <value>Prevent behavioral change</value>
```
Comment (buyaa-n) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1993]: Maybe better to mention that its related to the behavioral changes in IntPtr/UIntPtr, something like
```suggestion
    <value>Prevent numeric IntPtr/UIntPtr behavioral changes</value>
```

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx [hunk 2]
```diff
@@ -1981,7 +1981,7 @@
     <value>Prefer 'Clear' over 'Fill'</value>
   </data>
   <data name="PreventNumericIntPtrUIntPtrBehavioralChangesDescription" xml:space="preserve">
-    <value>Some built in operators added in .NET 7 behave differently than the user defined operatorsi in .NET 6 and below. Some operators that used to throw in unchecked context while overflowing will not throw anymore unless wrapped within checked context, and some operators that not used to throw in checked context now would throw unless wrapped within unchecked context.</value>
+    <value>Some built in operators added in .NET 7 behave differently when overflowing than did the user defined operators in .NET 6 and below. Some operators that used to throw in an unchecked context now no longer throw unless wrapped within a checked context, and some operators that did not previously throw in a checked context now throw unless wrapped in an unchecked context.</value>
```
Comment (carlossanlop) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1984]:
```suggestion
    <value>Some built in operators added in .NET 7 behave differently when overflowing than the user defined operators in .NET 6 and below did. Some operators that used to throw in an unchecked context now no longer throw unless wrapped within a checked context, and some operators that did not previously throw in a checked context now throw unless wrapped in an unchecked context.</value>
```

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx (hunk 1 above)
Comment (carlossanlop) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1993]:
```suggestion
    <value>Prevent behavioral change.</value>
```

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.cs.xlf [hunk 3]
```diff
@@ -2118,8 +2118,8 @@
...
-        <source>Some built in operators added in .NET 7 behave differently than the user defined operatorsi in .NET 6 and below. Some operators that used to throw in unchecked context while overflowing will not throw anymore unless wrapped within checked context, and some operators that not used to throw in checked context now would throw unless wrapped within unchecked context.</source>
-        <target state="new">Some built in operators added in .NET 7 behave differently than the user defined operatorsi in .NET 6 and below. Some operators that used to throw in unchecked context while overflowing will not throw anymore unless wrapped within checked context, and some operators that not used to throw in checked context now would throw unless wrapped within unchecked context.</target>
+        <source>Some built in operators added in .NET 7 behave differently when overflowing than did the user defined operators in .NET 6 and below. Some operators that used to throw in an unchecked context now no longer throw unless wrapped within a checked context, and some operators that did not previously throw in a checked context now throw unless wrapped in an unchecked context.</source>
```
Comment (carlossanlop) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.cs.xlf:2121]:
```suggestion
        <source>Some built in operators added in .NET 7 behave differently when overflowing than the user defined operators in .NET 6 and below did. Some operators that used to throw in an unchecked context now no longer throw unless wrapped within a checked context, and some operators that did not previously throw in a checked context now throw unless wrapped in an unchecked context.</source>
```

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.cs.xlf [hunk 4]
```diff
@@ -2128,8 +2128,8 @@
         <note />
       </trans-unit>
       <trans-unit id="PreventNumericIntPtrUIntPtrBehavioralChangesTitle">
-        <source>Prevent from behavioral change</source>
-        <target state="new">Prevent from behavioral change</target>
+        <source>Prevent behavioral change</source>
```
Comment (carlossanlop) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.cs.xlf:2131]:
```suggestion
        <source>Prevent behavioral change.</source>
```

**Review by carlossanlop — COMMENTED** [review]
I left some suggestions, they would need to get applied to the rest of the files.

@gewarren does my suggestion make sense, or would you word it differently?

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx (hunk 2 above)
Comment (gewarren) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:1984]: How are the built-in operators related to the user-defined operators? Would this make sense:
```suggestion
    <value>Some built-in operators added in .NET 7 behave differently when overflowing than did the corresponding user-defined operators in .NET 6 and earlier versions. Some operators that previously threw in an unchecked context now don't throw unless wrapped within a checked context. Also, some operators that did not previously throw in a checked context now throw unless wrapped in an unchecked context.</value>
```

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.cs.xlf (hunk 4 above)
Comment (buyaa-n) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.cs.xlf:2131]: Titles should not have a period

**Review by gewarren — APPROVED** [review]

**Review by jeffhandley — APPROVED** [review]

**Pull request merged**